Tests/smile_counter_dev.py

```diff
@@ -28,16 +24,29 @@ def _create_output_path(base_dir: str, filename: str, extension: str) -> str:
     return output_path
 
 
-def main():
-    # Logger config
-    logging.basicConfig(filename=LOG_OUTPUT_PATH, level=logging.INFO)
+def save_output_images(photo_name: str, img: cv2.Mat):
+    if config_data['SAVE_OUTPUT_IMAGES']:
+        filename = _create_output_path(config_data['DEFAULT_OUTPUT_FOLDER_PATH'], photo_name, config_data['DEFAULT_OUTPUT_FILE_EXTENSION'])
+        cv2.imwrite(filename, img)
+
+
+def show_output_images(window_name: str, img: cv2.Mat):
+    if config_data['SHOW_OUTPUT_IMAGES']:
+        cv2.imshow(window_name, img)
 
+
+def smile_detection(positive_dataset = True):
     # Load the cascade
-    smile_cascade = cv2.CascadeClassifier(SMILE_CLASSIFIER_PATH)
-    face_cascade = cv2.CascadeClassifier(FRONTAL_FACE_CLASSIFIER_PATH)
+    smile_cascade = cv2.CascadeClassifier(config_data['SMILE_CLASSIFIER_PATH'])
```

Comment:
I'm not sure if you need to use a reference to config_data[ ] when importing constants from configuration (it should be visible from the file level), but it's possible that it's caused by using .ini instead of .py.

In any case, it's something to check at least because these references clutter the code.

It's better to have "CONSTANT_NAME" instead of place_with_stored_constants["CONSTANT_NAME"]